packages/nimble-components/src/select/index.ts

```diff
@@ -49,51 +146,701 @@ export class Select extends FoundationSelect implements ErrorPattern {
+        );
+
+        if (selectedIndex !== -1) {
+            this.selectedIndex = selectedIndex;
+            return;
+        }
+
+        this.selectedIndex = 0;
+    }
+
+    /**
+     * Filter available options by text value.
+     *
+     * @public
+     */
+    private filterOptions(): void {
+        const filter = this.filter.toLowerCase();
+
+        if (filter) {
+            this.filteredOptions = this._options.filter(option => {
+                return diacriticInsensitiveStringNormalizer(
+                    option.text
+                ).includes(diacriticInsensitiveStringNormalizer(filter));
+            });
+        } else {
+            this.filteredOptions = this._options;
+        }
+
+        this._options.forEach(o => {
+            o.hidden = !this.filteredOptions.includes(o);
+        });
+    }
+
+    /**
+     * Sets the value and display value to match the first selected option.
+     *
+     * @param shouldEmit - if true, the input and change events will be emitted
+     *
+     * @internal
+     */
+    private updateValue(shouldEmit?: boolean): void {
+        if (this.$fastController.isConnected) {
+            this.value = this.firstSelectedOption?.value ?? '';
+        }
+
+        if (shouldEmit) {
+            this.$emit('input');
+            this.$emit('change', this, {
+                bubbles: true,
+                composed: undefined
+            });
         }
     }
 
+    /**
+     * Resets and fills the proxy to match the component's options.
+     *
+     * @internal
+     */
+    private setProxyOptions(): void {
+        if (this.proxy instanceof HTMLSelectElement && this.options) {
+            this.proxy.options.length = 0;
+            this.options.forEach(option => {
+                const proxyOption = option.proxy
+                    || (option instanceof HTMLOptionElement
+                        ? option.cloneNode()
+                        : null);
+
+                if (proxyOption) {
+                    this.proxy.options.add(proxyOption);
+                }
+            });
+        }
+    }
+
+    private clearSelection(): void {
+        this.options.forEach(option => {
+            option.selected = false;
+        });
+    }
+
+    private filterChanged(): void {
+        this.filterOptions();
+    }
+
     private maxHeightChanged(): void {
         this.updateListboxMaxHeightCssVariable();
     }
 
+    private initializeOpenState(): void {
+        if (!this.open) {
+            this.ariaExpanded = 'false';
+            this.ariaControls = '';
+            return;
+        }
+
+        this.committedSelectedOption = this._options[this.selectedIndex];
+        this.ariaControls = this.listboxId;
+        this.ariaExpanded = 'true';
+
+        this.setPositioning();
+        this.focusAndScrollOptionIntoView();
+    }
+
+    private focusFilterInput(): void {
+        if (this.open && this.filterMode !== FilterMode.none) {
+            window.requestAnimationFrame(() => {
```

Comment:
Do we have policies about using `requestAnimationFrame()` vs `DOM.queueUpdate()` or other update queue options? I can't recall but I know we don't use rAF very often so wanted to point it out.

Reply:
Yes, this has been a pain point with this implementation and has been discussed with @rajsite and @mollykreis independently. Ultimately, there was no other _practical_ known way to successfully get the focus transferred to the filter input (the other way that worked was a `DOM.queueUpdate()` wrapped in an `DOM.queueUpdate()`). We also found a usable within FAST that also did a rAF to accomplish a specific task (their `Listbox` implementation to scroll an element into view`), so it seemed to have precedent.